src/main/java/ui/MenuControl.java

```diff
@@ -173,21 +192,27 @@ private void onBoardSaveAs() {
      * Prompts a BoardNameDialog and saves the current board based on its response
      */
     public final void saveBoardAs() {
```

Comment:
Can you help to update header comment? Should not be tied to implementation. `Prompts for board name and save the current board as per the name` or something similar.